### PR TITLE
Docs master

### PR DESCRIPTION
**Pull request by kay-kim**

### CODE REVIEW

File: docs/reference/content/driver-async/getting-started/quick-tour-admin.md [hunk 1]
```diff
@@ -1,236 +0,0 @@
...
-```
-
-## Get A List of Collections
-
-You can get a list of the available collections in a database:
-
-```java
-database.listCollectionNames().forEach(new Block<String>() {
-    @Override
-    public void apply(final String databaseName) {
```
Comment (jyemin) [inline docs/reference/content/driver-async/getting-started/quick-tour-admin.md]: Change from databaseName to name or collectionName (name is used in the sync playbook)

File: docs/reference/content/driver-async/tutorials/databases-collections.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+```
+
+## Get A List of Collections
+
+You can get a list of the collections in a database using the [`MongoDatabase.listCollectionNames()`]({{<apiref "com/mongodb/async/client/MongoDatabase.html#listCollectionNames--">}}) method:
+
+```java
+database.listCollectionNames().forEach(new Block<String>() {
+    @Override
+    public void apply(final String databaseName) {
```
Comment (jyemin) [inline docs/reference/content/driver-async/tutorials/databases-collections.md]: Change databaseName to name

File: docs/reference/content/driver-async/tutorials/gridfs.md [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+The following example uploads into a `GridFSBucket` via the returned `GridFSUploadStream`:
+
+```java
+ByteBuffer data = ByteBuffer.wrap(
+                          "/tmp/mongodb-tutorial.pdf".getBytes(StandardCharsets.UTF_8);
+final GridFSUploadStream uploadStream = gridFSBucket.openUploadStream("sampleData");
+
+uploadStream.write(data, new SingleResultCallback<Integer>() {
+    @Override
+    public void onResult(final Integer result, final Throwable t) {
```
Comment (jyemin) [inline docs/reference/content/driver-async/tutorials/gridfs.md]: CC @rozza 

This is not guaranteed to work as the write method won't necessarily write all the bytes remaining in data.  In other words: 0 <= result <= data.remaining(), so for this method of writing the user may have to repeatedly call write until data.remaining == 0.

File: docs/reference/content/driver-async/tutorials/gridfs.md [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+The [`openDownloadStream`]({{< apiref "com/mongodb/async/client/gridfs/GridFSBucket.html#openDownloadStream-org.bson.types.ObjectId-">}}) method returns a [`GridFSDownloadStream`]({{< apiref "com/mongodb/client/gridfs/GridFSDownloadStream.html">}}) which extends [`AsyncInputStream`]({{< apiref "com/mongodb/async/client/gridfs/AsyncInputStream" >}}).
+
+The following example reads from the `GridFSBucket` via the returned `AsyncInputStream`. For the `fileId` variable, specify an `ObjectId` value returned from the find operation section in this tutorial.
+
+```java
+final ByteBuffer dstByteBuffer = ByteBuffer.allocate(1024 * 1024);
+final GridFSDownloadStream downloadStream = gridFSBucket.openDownloadStream(fileId);
+downloadStream.read(dstByteBuffer, new SingleResultCallback<Integer>() {
+    @Override
+    public void onResult(final Integer result, final Throwable t) {
```
Comment (jyemin) [inline docs/reference/content/driver-async/tutorials/gridfs.md]: CC @rozza 

Same problem here I believe: read may invoke the callback while there is still room is dstByteBuffer, so may have to call read repeatedly to fill it.

File: docs/reference/content/driver/tutorials/commands.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+For additional information on connecting to MongoDB, see [Connect to MongoDB]({{< ref "connect-to-mongodb.md">}}).
+
+## Run `buildInfo` and `collStats` Commands
+
+To run a command, construct a [`Document`]({{< apiref "org/bson/Document.html" >}})
+object that specifies the command and pass it to the `runCommand()` method.
+
+The following runs the [`buildInfo`]({{<docsref "reference/command/buildInfo">}}) command and the [`collStats`]({{<docsref "reference/command/collStats">}}):
```
Comment (jyemin) [inline docs/reference/content/driver/tutorials/commands.md]: "and the collStats command"

File: docs/reference/content/driver-async/tutorials/commands.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+For additional information on connecting to MongoDB, see [Connect to MongoDB]({{< relref "driver-async/tutorials/connect-to-mongodb.md">}}).
+
+## Run `buildInfo` and `collStats` Commands
+
+To run a command, construct a [`Document`]({{< apiref "org/bson/Document.html" >}})
+object that specifies the command and pass it to the `runCommand()` method.
+
+The following runs the [`buildInfo`]({{<docsref "reference/command/buildInfo">}}) command and the [`collStats`]({{<docsref "reference/command/collStats">}}):
```
Comment (jyemin) [inline docs/reference/content/driver-async/tutorials/commands.md]: "and the collStats command"

**Comment by jyemin** [comment]
Thanks Kay.  It's looking great.

**Comment by jyemin** [comment]
Two remaining issues:

1.  I recommend removing the openUploadStream/openDownloadStream examples for the initial version.  The code is going to get rather complicated if we want to do it right.  We can just mention the existence of the methods and say that they are there if you require fine-grained control of the upload/download process.

2. Please run 'gradle -x test check' and fix the checkstyle issues in the new Java examples  (that's why the Travis build is failing)

**Comment by kay-kim** [comment]
Updated the gridfs blurbs : )  
Also, ran gradle and found that I had accidentally committed the little java class I was using to test things as I was writing.  So removed that little sucker.

**Pull request closed without merging**